sw/host/tests/chip/spi_device_tpm_test/src/main.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Result};
+use clap::Parser;
+use regex::Regex;
+use std::path::PathBuf;
+use std::time::Duration;
+
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::execute_test;
+use opentitanlib::test_utils::init::InitializeTest;
+use opentitanlib::tpm::{self, Driver};
+use opentitanlib::uart::console::{ExitStatus, UartConsole};
+
+#[derive(Debug, Parser)]
+struct Opts {
+    #[command(flatten)]
+    init: InitializeTest,
+
+    /// Console receive timeout.
+    #[arg(long, value_parser = humantime::parse_duration, default_value = "600s")]
+    timeout: Duration,
+
+    /// Name of the SPI interface to connect to the OTTF console.
+    #[arg(long, default_value = "TPM")]
+    spi: String,
+
+    /// Path to the firmware's ELF file, for querying symbol addresses.
+    #[arg(value_name = "FIRMWARE_ELF")]
+    firmware_elf: PathBuf,
+}
+
+fn tpm_read_test(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
+    let uart = transport.uart("console")?;
+    let spi = transport.spi(&opts.spi)?;
+    transport.pin_strapping("SPI_TPM")?.apply()?;
+
+    /* Wait sync message. */
+    let _ = UartConsole::wait_for(&*uart, r"SYNC: Begin TPM Test\r\n", opts.timeout)?;
+    let tpm = tpm::SpiDriver::new(spi);
+    const SIZE : usize = 10;
+    
+    for _ in 0..10 {
+        let test_data : Vec<u8> = (0..SIZE).map(|_| rand::random()).collect();
```

Comment:
This could probably be punted to another PR, but we might want randomly-generated stimulus to be provided by some PRNG that is fed by a random (but controllable) seed. Otherwise, failures can become a bit more annoying to reproduce.